Assets/Scripts/SS3D/Systems/Entities/Silicon/ThreadController.cs

```diff
@@ -130,9 +133,13 @@ private void MovePlayer()
         /// <returns></returns>
         private void ProcessPlayerInput()
         {
-            float x = _controls.Movement.ReadValue<Vector2>().x;
-            float y = _controls.Movement.ReadValue<Vector2>().y;
-
+            float x = 0;
+            float y = 0;
+            if (_controls.Movement.phase == InputActionPhase.Performed)
```

Comment:
can you explain what's "phase" and InputActionPhase ?

Reply:
Every action switches between five phases: Disabled, Waiting, Started, Perfromed, Canceled. For example, if you have hold(0.5 seconds) interaction on an action, the action has performed phase when you hold a key for 0.5 seconds or more